crashtracker/src/crash_handler.rs

```diff
@@ -142,6 +158,17 @@ extern "C" fn handle_posix_signal(signum: i32) {
     // return to old handler (chain).  See comments on `restore_old_handler`.
 }
 
+fn emit_metadata(w: &mut impl Write) -> anyhow::Result<()> {
+    writeln!(w, "{DD_CRASHTRACK_BEGIN_METADATA}")?;
+
+    let metadata = unsafe { &METADATA_STRING.as_ref().context("Expected metadata")? };
+    writeln!(w, "{}", metadata)?;
+
+    writeln!(w, "{DD_CRASHTRACK_END_METADATA}")?;
+
+    Ok(())
+}
+
 fn handle_posix_signal_impl(signum: i32) -> anyhow::Result<()> {
     let mut receiver = match std::mem::replace(unsafe { &mut RECEIVER }, GlobalVarState::Taken) {
         GlobalVarState::Some(r) => r,
```

Comment:
One thing that occurred to me while reviewing the PR is -- should we have something protecting `handle_posix_signal_impl` so that it can't get called multiple times in the same process?

Afaik if a thread is handling a signal (e.g. `SIGSEGV`), no other SIGSEGVs can be delivered. But I... don't think that applies to other threads in the same process.

May be worth giving it a try?

Reply:
https://github.com/DataDog/libdatadog/issues/318 to track